src/Knp/FriendlyContexts/Guesser/IntGuesser.php

```diff
@@ -18,7 +18,31 @@ public function transform($str, array $mapping = null)
 
     public function fake(array $mapping)
     {
-        return current($this->fakers)->fake('numberBetween', [0, 2000000000]);
+        $min = 0;
+        $max = $this->determineMaxValue($mapping);
+
+        return current($this->fakers)->fake('numberBetween', [$min, $max]);
+    }
+
+    /**
+     * @param array $mapping
+     *
+     * @return int
+     */
+    private function determineMaxValue(array $mapping)
+    {
+        $defaultMax = 2000000000;
+
+        if (!isset($mapping['length']) || $mapping['length'] < 1) {
```

Comment:
i think you could also check `|| $mapping['length'] >= 10` here and then just do `return = (int)str_repeat('9', $mapping['length']);` below?